### PR TITLE
[2.6 docs cleanup] Make schema evolution more unified

### PR DESCRIPTION
**Pull request by helena**
One idea of bubbling up the larger topic of schema evolution by consolidation of what is currently scattered in a few places for their specific solutions. These share many similarities  
and there is overlap and some duplication of content. After thinking about this for a while looking at all the related docs, it began to make more sense in one file, and I can see how over time we might want to make schema evolution in general **a more unified front**.
https://github.com/akka/akka/pull/27551
* Part of #27223 with crossover for 2.6 and Typed
* Further re-org

- [ ] TODO fix why new `schema-evolution.md` is not showing up as a new section in the Package Deploy Rolling Updates of `deploy.md`

### CODE REVIEW

**Review by johanandren — COMMENTED** [review]
I'm not convinced this is the direction to take it, the biggest problem I have seen related to evolution with users/customers is that they have missed out on thinking about it for persistence, users that wants to do rolling upgrades have to a much larger extent already been on a level where they already got it. Given that I think that we shouldn't de-emphasis persistence by talking more about rolling upgrades or turning the texts about it more generic.

**Comment by helena** [comment]
I'm not convinced either, can argue for both, but thought it worth raising the question.

**Comment by helena** [comment]
The thing is, schema evolution is a larger topic, and in these docs it's raised in multiple places where I'd found duplication / similarities which doesn't seem ideal. Yes, it's a persistence issue  but it's also related to rolling updates and serialization concerns.

File: akka-docs/src/main/paradox/serialization-jackson.md [hunk 1]
```diff
@@ -173,179 +173,6 @@ Scala
...
-When a class is not used for serialization any more it can be removed from `serialization-bindings` but to still
-allow deserialization it must then be listed in the `whitelist-class-prefix` configuration. This is useful for example
-during rolling update with serialization changes, or when reading old stored data. It can also be used
-when changing from Jackson serializer to another serializer (e.g. Protobuf) and thereby changing the serialization
-binding, but it should still be possible to deserialize old data with Jackson.
-
-@@snip [config](/akka-serialization-jackson/src/test/scala/doc/akka/serialization/jackson/SerializationDocSpec.scala) { #whitelist-class-prefix }
-
-It's a list of class names or prefixes of class names.
-
```
Comment (patriknw) [inline akka-docs/src/main/paradox/serialization-jackson.md]: if we move this out of here we must include a link

**Review by patriknw — COMMENTED** [review]
I like the intent of placing everything together. The section Event Sourced Actors is what requires an overhaul since some of it is outdated (or we have different opinions nowadays) and the event adapter samples should be updated using Akka Typed APIs which has a simplified form of event adapters.

Not sure if "Schema evolution" is still the right name of this. I personally associate it with databases. "Serialization format evolution"?

**Pull request closed without merging**